tests/unit/locking_test.cc

```diff
@@ -420,3 +422,170 @@ SEASTAR_THREAD_TEST_CASE(test_with_lock_typed_return_throwing_move_func) {
         }
     }
 }
+
+SEASTAR_TEST_CASE(test_shared_mutex_locks) {
+    shared_mutex sm;
+
+    {
+        const auto ulock = co_await get_unique_lock(sm);
+        BOOST_REQUIRE(!sm.try_lock());
+        BOOST_REQUIRE(!sm.try_lock_shared());
+    }
+
+    BOOST_REQUIRE(sm.try_lock());
+    sm.unlock();
+
+    {
+        const auto slock1 = co_await get_shared_lock(sm);
+        BOOST_REQUIRE(!sm.try_lock());
+        BOOST_REQUIRE(sm.try_lock_shared());
+
+        const auto slock2 = co_await get_shared_lock(sm);
+
+        // This balances out the call to `try_lock_shared()` above.
+        sm.unlock_shared();
+    }
+
+    BOOST_REQUIRE(sm.try_lock());
+    sm.unlock();
+}
+
+SEASTAR_TEST_CASE(test_shared_mutex_exclusive_locks) {
+    shared_mutex sm{};
+    unsigned counter = 0;
+
+    co_await coroutine::parallel_for_each(boost::irange(0, 10), coroutine::lambda([&sm, &counter] (auto&&) -> future<> {
+        const auto ulock = co_await get_unique_lock(sm);
+        BOOST_REQUIRE_EQUAL(counter, 0u);
+        ++counter;
+        co_await sleep(1ms);
+        --counter;
+        BOOST_REQUIRE_EQUAL(counter, 0u);
+    }));
+}
+
+SEASTAR_TEST_CASE(test_shared_mutex_shared_locks) {
+    shared_mutex sm{};
+    unsigned counter = 0;
+    unsigned max = 0;
+
+    co_await coroutine::parallel_for_each(boost::irange(0, 10),
+            coroutine::lambda([&sm, &counter, &max] (auto&&) -> future<> {
+        const auto slock = co_await get_shared_lock(sm);
+        ++counter;
+        max = std::max(max, counter);
+        co_await sleep(1ms);
+        --counter;
+    }));
+
+    BOOST_REQUIRE_EQUAL(counter, 0u);
+    BOOST_REQUIRE_NE(max, 0u);
```

Comment:
I don't think this last check checks anything - obviously max can't be 0 because it will always be at least 1 (you increment counter, so it's at least 1, and then set max to at least that).

I'm not sure you really needed to check all this parallelization stuff, I think test_shared_mutex_locks above already checks the same thing more simply (it tries to take the same lock 3 times, and it succeeds).

Reply:
Can you please answer this?
It's not a critical issue (who cares if a test checks a silly condition that can never be false), but it is a hint that maybe you missed the fact that the test isn't testing what you think it checks - so maybe you forgot to test what you really wanted to test, or something. So please just have a look.

Reply:
I'm sorry for the late response. Most of the tests I added are analogues of the existing tests with using locks and coroutines being the only difference. I think you're right it's not necessary to check those things again. You're definitely right that comparing `max` against 0 is redundant. It would make more sense if Seastar were preemptive. I'll remove the test.